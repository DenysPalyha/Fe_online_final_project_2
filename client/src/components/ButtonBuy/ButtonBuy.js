// eslint-disable-next-line no-unused-vars
import React from 'react';
import PropTypes from 'prop-types';
import styles from './ButtonBuy.module.scss';

function ButtonBuy(props) {
  const {
    padding, backgroundColor, handleClick, text,
  } = props;
  return (
    <button type="button" className={styles.buttonsMain} style={{ backgroundColor, padding }} onClick={handleClick}>
      {text}
    </button>
  );
}

ButtonBuy.propTypes = {
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  backgroundColor: PropTypes.string,
  padding: PropTypes.string,
  handleClick: PropTypes.func,
};

ButtonBuy.defaultProps = {
  backgroundColor: '#456F49',
  padding: '15px 30px',
  handleClick: () => {},
};

export default ButtonBuy;
