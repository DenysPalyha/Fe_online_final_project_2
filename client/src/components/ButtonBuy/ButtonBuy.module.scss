.buttonsMain {
  font-size: 16px;
  margin-top: 12px;
  border-radius: 5px;
  border: none;
  outline: none;
  color: #fff;
  text-shadow: 0 1px 0 #b6e6f9;

  &:hover {
    cursor: pointer;
    filter: drop-shadow(4px 4px 4px #777) contrast(125%);
  }
}
