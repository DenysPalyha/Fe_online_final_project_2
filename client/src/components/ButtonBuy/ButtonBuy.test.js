import { render } from '@testing-library/react';
import ButtonBuy from './ButtonBuy';
// eslint-disable-next-line no-undef
describe('ButtonBuy snapshot testing', () => {
  // eslint-disable-next-line no-undef
  test('should ButtonBuy match snapshot', () => {
    const { asFragment } = render(<ButtonBuy text="some text" />);
    // eslint-disable-next-line no-undef
    expect(asFragment()).toMatchSnapshot();
  });
});
