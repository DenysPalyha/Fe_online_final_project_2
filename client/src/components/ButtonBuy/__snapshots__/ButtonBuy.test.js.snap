// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBuy snapshot testing should ButtonBuy match snapshot 1`] = `
<DocumentFragment>
  <button
    class="buttonsMain"
    style="background-color: rgb(69, 111, 73); padding: 15px 30px;"
    type="button"
  >
    some text
  </button>
</DocumentFragment>
`;
